azurerm/internal/services/databasemigration/tests/resource_arm_database_migration_service_test.go

```diff
@@ -0,0 +1,233 @@
+package azurerm
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/features"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/databasemigration/parse"
+
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/acceptance"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func TestAccAzureRMDatabaseMigrationService_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_database_migration_service", "test")
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMDatabaseMigrationServiceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMDatabaseMigrationService_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMDatabaseMigrationServiceExists(data.ResourceName),
+					resource.TestCheckResourceAttrSet(data.ResourceName, "subnet_id"),
+					resource.TestCheckResourceAttr(data.ResourceName, "sku_name", "Standard_1vCores"),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+func TestAccAzureRMDatabaseMigrationService_complete(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_database_migration_service", "test")
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMDatabaseMigrationServiceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMDatabaseMigrationService_complete(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMDatabaseMigrationServiceExists(data.ResourceName),
+					resource.TestCheckResourceAttrSet(data.ResourceName, "subnet_id"),
+					resource.TestCheckResourceAttr(data.ResourceName, "sku_name", "Standard_1vCores"),
+					resource.TestCheckResourceAttr(data.ResourceName, "tags.name", "test"),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
```

Comment:
Could we also do an update test here where we go from basic -> complete -> basic?